test/linux/mbox-server-tester.c

```diff
@@ -0,0 +1,109 @@
+#define _GNU_SOURCE
+
+#include <errno.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <sched.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "libmbox/mbox.h"
+#include "mailbox-map.h"
+#include "mbox-utils.h"
+
+#define PATH_SIZE 128
+
+#define RTPS_PATH_IN "/dev/mbox/1/mbox4"
+#define RTPS_PATH_OUT "/dev/mbox/1/mbox3"
+
+
+enum mbox_notif notif_type = MBOX_NOTIF_NONE;
+int timeout_ms = -1;
+
+static const char short_options[] = "n:t:c:h";
+static const struct option long_options[] = {
+    {"not-type",    required_argument,  NULL,   'n'},
+    {"timeout",     required_argument,  NULL,   't'},
+    {"cpu",         required_argument,  NULL,   'c'},
+    {"help",        no_argument,        NULL,   'h'},
+    {0, 0, 0, 0}
+};
+
+int main(int argc, char** argv)
+{
+	struct mbox mbox_in;
+	struct mbox mbox_out;
+
+	int cpu = -1;
+
+	int c;	
+	while ((c = getopt_long(argc, argv, short_options, long_options, NULL)) != -1) {
+        switch (c) {
+        case 'n':
+            if (!strcmp(optarg, "none")) {
+                notif_type = MBOX_NOTIF_NONE;
+            } else if (!strcmp(optarg, "select")) {
+                notif_type = MBOX_NOTIF_SELECT;
+            } else if (!strcmp(optarg, "poll")) {
+                notif_type = MBOX_NOTIF_POLL;
+            } else if (!strcmp(optarg, "epoll")) {
+                notif_type = MBOX_NOTIF_EPOLL;
+            } else {
+                fprintf(stderr, "Unknown notification type: %s\n", optarg);
+                //usage(argv[0], EINVAL);
+            }
+            break;
+        case 't':
+            timeout_ms = atoi(optarg);
+            break;
+        case 'c':
+            cpu = atoi(optarg);
+            break;
+    	}
```

Comment:
Please add an additional case for 'h', the help menu.  Right now, you include it in the short_options and long_options arrays in lines 26 and 27-33, but you don't support the actual functionality here.  You may need to create a usage() function (like in mboxtester.c) that can be called if the user passes in the help flag.  In addition, if the user passes in a flag without a required argument, don't execute the server- instead, tell the user that the required arg is missing, then execute the usage() function to show them how to pass arguments to your code.